src/shared_memory_manager.cc

```diff
@@ -346,12 +346,26 @@ SharedMemoryManager::GetMemoryInfo(
         std::string("Unable to find shared memory region: '" + name + "'")
             .c_str());
   }
+
+  // populate 'shm_mapped_addr'
+  size_t max_offset = 0;
```

Comment:
Small suggestion - what if we refactored to set the offsets, check them, and then only set the shm variable? I think one thing was a little odd was seeing the address set and then the offsets checked. 

By setting the variables you can also keep a single place to set the address - just a thought.

```
size_t max_offset = 0;
size_t offser = 0;
 
if (it->second->byte_size_ > 0) {
    max_offset = it->second->byte_size_ - 1;
  }
  if (it->second->kind_ == TRITONSERVER_MEMORY_CPU) {
     offset = it->second->offset + offset;
     max_offset += it -> second-> offset;
  } 
 
  if (offset > max_offset) {
    *shm_mapped_addr = nullptr;
    return TRITONSERVER_ErrorNew(
        TRITONSERVER_ERROR_INVALID_ARG,
        std::string("Invalid offset for shared memory region: '" + name + "'")
            .c_str());
  }
   *shm_mapped_addr = (void*)((uint8_t*)it->second->mapped_addr_ + offset);
```

Reply:
Sure. [Separate logic between computation and validation](https://github.com/triton-inference-server/server/pull/6914/commits/c750fdf9f4b2bda12b359d219d76f02900528ce1)